src/commands/commerce/extension/map.ts

```diff
@@ -0,0 +1,127 @@
+    public async run(): Promise<string> {
+        this.ux.log(`Accessing Store using username: ${this.org.getUsername()} \n..........`);
+
+        return this.processMapExtension(
+            this.flags['registered-extension-name'],
+            this.flags['store-name'],
+            this.flags['store-id'],
+            this.org.getUsername()
+        );
+    }
+
+    public processMapExtension(extensionName: string, storeName: string, storeId: string, userName: string): string {
+        const storeid = this.getStoreId(storeName, storeId, userName);
+        const registeredExternalServiceId = this.getExtensionName(extensionName, userName);
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const results = forceDataRecordCreate(
+            'StoreIntegratedService',
+            `Integration=${registeredExternalServiceId} StoreId=${storeid} ServiceProviderType='Extension'`,
+            userName
+        );
+
+        if (results instanceof SfdxError) {
+            throw new SfdxError(msgs.getMessage('extension.map.error', [extensionName, results.message]));
+        }
+        // JSON response of inserted record
+        return this.getInsertedRecord(storeid, registeredExternalServiceId);
+    }
+
+    private getExtensionName(extensionName: string, userName: string): string {
+        let registeredExternalServiceId: string;
+        try {
+            registeredExternalServiceId = forceDataSoql(
+                `SELECT Id FROM RegisteredExternalService WHERE DeveloperName='${extensionName}'`,
+                userName
+            ).result.records[0].Id;
+        } catch (error) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            const errorMsg = msgs.getMessage('extension.map.nonexistant', [error.message]);
+            throw new SfdxError(errorMsg);
+        }
+        return registeredExternalServiceId;
+    }
+
+    private getStoreId(storeName: string, storeId: string, userName: string): string {
+        if (storeId === undefined) {
+            try {
+                storeId = forceDataSoql(`SELECT Id FROM WebStore WHERE Name='${storeName}' LIMIT 1`, userName).result
+                    .records[0].Id;
+            } catch (e) {
+                throw new SfdxError(
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                    msgs.getMessage('extension.map.errStoreName', [storeName, e.message])
+                );
+            }
+        } else {
+            try {
+                storeId = forceDataSoql(`SELECT Id FROM WebStore WHERE Id='${storeId}' LIMIT 1`, userName).result
+                    .records[0].Id;
+            } catch (e) {
+                throw new SfdxError(
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                    msgs.getMessage('extension.map.errStoreId', [storeId, e.message])
```

Comment:
Could we add a `/n` before e.message here and on line 94 so the error message will output on the next line rather than the same line as the generic message?